alicloud/resource_alicloud_yundun_dbaudit.go

```diff
@@ -0,0 +1,204 @@
+package alicloud
+
+import (
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/yundun_dbaudit"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/bssopenapi"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+const (
+	RELEASE_HANG_MINS = 3
+)
+
+func resourceAlicloudDbauditInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudDbauditInstanceCreate,
+		Read:   resourceAlicloudDbauditInstanceRead,
+		Update: resourceAlicloudDbauditInstanceUpdate,
+		Delete: resourceAlicloudDbauditInstanceDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateStringLengthInRange(1, 64),
+			},
+			"plan_code": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"period": {
+				Type:         schema.TypeInt,
+				ValidateFunc: validateAllowedIntValue([]int{1, 3, 6, 12, 24, 36}),
+				Optional:     true,
+				Default:      1,
+			},
+			"vswitch_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudDbauditInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	request := buildDbauditCreateRequest(d, meta)
+	raw, err := client.WithBssopenapiClient(func(bssopenapiClient *bssopenapi.Client) (interface{}, error) {
+		return bssopenapiClient.CreateInstance(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_yundun_dbaudit_instance", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+
+	response := raw.(*bssopenapi.CreateInstanceResponse)
+	instanceId := response.Data.InstanceId
+	if !response.Success {
+		return WrapError(Error(response.Message))
+	}
+	d.SetId(instanceId)
+
+	dbauditService := DbauditService{client}
+
+	// check RAM policy
+	dbauditService.ProcessRolePolicy()
+	// set description
+	if err = dbauditService.UpdateDbauditInstanceDescription(instanceId, d.Get("description").(string)); err != nil {
```

Comment:
Without "Update" here, the configuration of "description" will not work as soon as the instance has been created. It might took the user another "Update" operation to make that configuration take effect

Reply:
will do